src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -46,40 +68,94 @@
         this.snapshotTimestampMs = snapshotTimestamp * 1000;
     }
 
+    /**
+     * Does two things
+     * <ol>
+     *     <li>Sets the minimum weight magnitude. We validate POW on a transaction we will ascertain that it has
+     *     a certain amount of 9s at the end of its hash. This amount depends on MWM</li>
+     *     <li>Starts the transaction solidification thread</li>
+     * </ol>
+     *
+     *
+     * @see #spawnSolidTransactionsPropagation()
+     * @param testnet true if we are in testnet mode, this caps {@code mwm} to {@value #TESTNET_MWM_CAP}
+     *                regardless of parameter input.
+     * @param mwm minimum weight magnitude: the minimal number of 9s that ought to appear in the end of the transaction
+     *            hash
+     */
     public void init(boolean testnet, int mwm) {
         setMwm(testnet, mwm);
 
         newSolidThread = new Thread(spawnSolidTransactionsPropagation(), "Solid TX cascader");
         newSolidThread.start();
     }
 
+    //Package Private For Testing
     void setMwm(boolean testnet, int mwm) {
         minWeightMagnitude = mwm;
 
         //lowest allowed MWM encoded in 46 bytes.
         if (!testnet){
-            minWeightMagnitude = Math.max(minWeightMagnitude, 13);
+            minWeightMagnitude = Math.max(minWeightMagnitude, TESTNET_MWM_CAP);
         }
     }
 
+    /**
+     * Shutdown roots to tip solidification thread
+     * @throws InterruptedException
+     * @see #spawnSolidTransactionsPropagation()
+     */
     public void shutdown() throws InterruptedException {
         shuttingDown.set(true);
         newSolidThread.join();
     }
 
+    /**
+     * @return the number of trailing 9s that have to be present in the transaction hash to validate that proof of work
```

Comment:
```suggestion
     * @return the number of trailing 9s that have to be present in the transaction hash to validate that enough proof of work
```?

Reply:
changed to: the minimal number of consecutive trailing 9s that have to be present at the end of the transaction hash in order to validate that sufficient proof of work has been done